dashboard/lib/services/lti.rb

```diff
@@ -30,7 +30,7 @@ def self.initialize_lti_user(id_token)
     end
 
     def self.create_lti_user_identity(user)
-      auth_option = user.authentication_options.find(&:lti?)
+      auth_option = user.authentication_options.order(created_at: :desc).find(&:lti?)
```

Comment:
Since order(created_at: :desc) triggers a DB request with ordering even if records are already loaded,

it's better to either perform the entire operation in a single query:
```suggestion
      auth_option = user.authentication_options.order(created_at: :desc).find_by(credential_type: AuthenticationOption::LTI_V1)
```

or sort on the ruby side if the records are already in memory:
```suggestion
      auth_option = user.authentication_options.select(&:lti?).max_by(&:created_at)
```